ERCS/erc-7518.md

```diff
@@ -0,0 +1,606 @@
+* Complex Revenue Sharing: By customizing revenue-sharing percentages for different token classes, the hotel can implement more intricate revenue-sharing models. For example, "Class A" tokens might have a higher revenue-sharing percentage due to the premium amenities and high demand on Floor 1, while "Class C" tokens for Floor 3, representing fractional ownership, might have a lower revenue-sharing percentage.
+* Optimized Revenue Distribution: By strategically adjusting the payout structures for various partitions, the hotel can optimize revenue distribution among token holders. This approach ensures that investors with tokens representing more desirable floors or premium features receive a greater share of the revenue, while still providing value to holders of tokens for less popular or less lucrative partitions.
+* Incentivizing Investments: Different revenue-sharing arrangements for token classes can incentivize investors to acquire tokens for specific floors. If a particular floor is expected to have higher revenue potential in the future, the corresponding tokens may attract more investments due to their attractive payout structure.
+* Catering to Stakeholder Preferences: The ability to create multiple token classes, each with its own unique payout structure, allows the hotel to cater to the diverse preferences of stakeholders. Some investors might be interested in higher-risk, higher-reward investments (e.g., "Class A" tokens), while others might prefer more conservative options (e.g., "Class C" tokens).
+
+By combining dynamic pricing, designated payouts, complex revenue sharing, optimized revenue distribution, incentivized investments, and stakeholder catering, the hotel can create a versatile and attractive investment opportunity that appeals to a broad range of investors. Moreover, the transparency and immutability of blockchain-based [ERC-1155](./eip-1155.md) tokens ensure a secure and trustless system for managing ownership, access, and revenue-sharing rights within the hotel's ecosystem.
+
+### Transfer Flow
+
+The transfer flow in the proposed EIP consists of two main parts off-chain and on-chain.
+
+* Client: Represents the client or user initiating the transfer. It interacts with the backend and the [ERC-7518](./eip-7518.md) contract.
+* Backend: Refers to the backend system responsible for handling requests from the client and performing off-chain verification.
+* [ERC-7518](./eip-7518.md): Represents the smart contract that implements the [ERC-7518](./eip-7518.md) security standard. It receives transfer requests and interacts with the compliance contract for verification.
+* Compliance: Responsible for performing verification checks. It verifies the transfer's validity based on various factors such as off-chain verification, custom rules, regulations, and signature validation.
+
+```mermaid
```

Comment:
I doubt mermaid works in our renderer. I'd recommend using an SVG diagram instead.